app/extlinux_control.py

```diff
@@ -474,45 +336,19 @@ def is_current_slot_marked_successful(self) -> bool:
     def reboot(cls):
         subprocess.check_call(shlex.split("reboot"))
 
-    def gen_extlinux_cfg(self, f: Path = None) -> str:
-        """
-        create extlinux config for standby slot based on imported cfg
-        """
-        _cboot = "${cbootargs} quiet "
-        _rootfs = f"root={self.standby_slot_partuuid} rw rootwait rootfstype=ext4"
-        _cmdline = f"{_cboot} {_rootfs} {self._cmdline_extra}"
-
-        # first try to load existed cfg files
-        # NOTE: not loading external extlinux.cfg rightnow
-        cfg = ExtlinuxCfgFile()
-        cfg.load_entry(
-            label="primary",
-            menu_lable="primary kernel",
-            linux=self._linux,
-            initrd=self._initrd,
-            fdt=self._fdt,
-            append=_cmdline,
-        )
-        cfg.set_default_entry("primary")
+    def update_extlinux_cfg(self, dst: Path, src: Path):
+        def _replace(ma: re.Match, repl: str):
+            append_l = ma.group(0)
+            if append_l.startswith("#"):
+                return append_l
+            res, n = re.compile(r"root=[\w\-=]*").subn(repl, append_l)
+            if not n:
+                res = f"{append_l} {repl}"
 
-        return cfg.dump_cfg()
+            return res
 
-    def write_extlinux_cfg(self, target: Path, src: Path = None):
-        """
-        should only be used to generate extlinux conf for standby slot
-        DO NOT TOUCH THE CURRENT SLOT'S EXTLINUX FILE!
-
-        write new extlinux conf file to target
-        """
-        if src and src.is_file():
-            # load template extlinux_cfg from external
-            cfg_text = self.gen_extlinux_cfg(src)
-        else:
-            # generate extlinux from default settings
-            cfg_text = self.gen_extlinux_cfg()
-
-        with open(target, "w") as f:
-            f.write(cfg_text)
+        _repl_func = partial(_replace, repl=self.standby_slot_partuuid)
```

Comment:
```suggestion
        _repl_func = partial(_replace, repl=f"root=PARTUUID={self.standby_slot_partuuid}")
```
Sorry I forgot to prepare the pure PARTUUID into "root=PARTUUID=<PARTUUID>" in my previous comment.

Reply:
Yes I confirm that!